test/format/import.c

```diff
@@ -0,0 +1,212 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "format.h"
+
+static void copy_file_into_directory(WT_SESSION *session, const char *dir, const char *name);
+static void import_with_repair(WT_SESSION *session);
+static void import_with_file_metadata(WT_SESSION *session, WT_SESSION *import_session);
+static void verify_import(WT_SESSION *session, int start_value);
+
+/*
+ * Import directory initialize command, remove and re-create the primary backup directory, plus a
+ * copy we maintain for recovery testing.
+ */
+#define HOME_IMPORT_INIT_CMD "rm -rf %s/" IMPORT_DIR "&& mkdir %s/" IMPORT_DIR
+#define IMPORT_DIR "IMPORT"
+#define IMPORT_URI "table:import"
+#define IMPORT_URI_CONFIG "key_format=i,value_format=i"
+#define IMPORT_ENTRIES 1000
+
+/*
+ * import --
+ *     Periodically import table.
+ */
+WT_THREAD_RET
+import(void *arg)
+{
+    WT_CONNECTION *conn, *import_conn;
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_SESSION *import_session, *session;
+    size_t cmd_len;
+    uint32_t import_value;
+    u_int period;
+    int counter;
+    char cmd[64];
+
+    WT_UNUSED(arg);
+    conn = g.wts_conn;
+    counter = 0;
+    import_value = false;
+
+    memset(cmd, 0, sizeof(cmd));
+    cmd_len = strlen(g.home) * 2 + strlen(HOME_IMPORT_INIT_CMD) + 1;
+    testutil_check(__wt_snprintf(cmd, cmd_len, HOME_IMPORT_INIT_CMD, g.home, g.home));
+    testutil_checkfmt(system(cmd), "%s", "import directory creation failed");
+
+    memset(cmd, 0, sizeof(cmd));
+    cmd_len = strlen(g.home) + strlen(IMPORT_DIR) + 10;
+    testutil_check(__wt_snprintf(cmd, cmd_len, "%s/%s", g.home, IMPORT_DIR));
+    /* Open a connection to the database, creating it if necessary. */
+    create_database(cmd, &import_conn);
+
+    /* Open a session */
+    testutil_check(import_conn->open_session(import_conn, NULL, NULL, &import_session));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    testutil_checkfmt(
+      import_session->create(import_session, IMPORT_URI, IMPORT_URI_CONFIG), "%s", g.uri);
+
+    testutil_check(import_session->open_cursor(import_session, IMPORT_URI, NULL, NULL, &cursor));
+    while (!g.workers_finished) {
+        period = mmrand(NULL, 1, 10);
+
+        for (int i = 0; i < IMPORT_ENTRIES; ++i) {
+            cursor->set_key(cursor, i);
+            cursor->set_value(cursor, counter + i);
+            testutil_check(cursor->insert(cursor));
+        }
+        counter += IMPORT_ENTRIES;
+        testutil_check(import_session->checkpoint(import_session, NULL));
```

Comment:
It occurs to me that we really want to be testing the act of importing into the test database. I am not sure we gain much by updating the import database table each iteration. I think you could simplify by moving the setup of the import table and populating it, then closing the cursor and `import_conn` outside the loop. Grab the metadata outside the loop for the `import_with_file_metadata` call. Then in the loop you basically start at the `copy_file_into_directory` piece. That simplifies it considerably and simplifies the verification piece too.

Let me know if you think the modification of the import table each iteration adds value.

Reply:
I like this modification, as it purely tests import